polaris/hub/client.py

```diff
@@ -587,13 +590,17 @@ def upload_dataset(
         if dataset.zarr_root is not None:
             with tmp_attribute_change(self.settings, "default_timeout", timeout):
                 # Copy the Zarr archive to the hub
-                # This does not copy the consolidated data
                 dest = self.open_zarr_file(
                     owner=dataset.owner,
                     name=dataset.name,
                     path=dataset_json["zarrRootPath"],
                     mode="w",
                 )
+                # Locally consolidate the archive metadata
+                zarr.consolidate_metadata(dataset.zarr_root.store)
+                zmetadata_content = dataset.zarr_root.store[".zmetadata"]
+                dest.store[".zmetadata"] = zmetadata_content
```

Comment:
Let's reference https://github.com/zarr-developers/zarr-python/issues/1731 in the docs here! This is a simple enough solution. My worry is that one day they might change how a consolidated archive is saved and we wouldn't notice it until things start breaking. Once there is an answer on that issue, we can update this code to reflect the recommendation from the Zarr developers.